cmd/kubelet/app/options/options.go

```diff
@@ -185,8 +214,6 @@ func (c *kubeletConfiguration) addFlags(fs *pflag.FlagSet) {
 		"If --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key "+
 		"are generated for the public address and saved to the directory passed to --cert-dir.")
 	fs.StringVar(&c.TLSPrivateKeyFile, "tls-private-key-file", c.TLSPrivateKeyFile, "File containing x509 private key matching --tls-cert-file.")
-	fs.StringVar(&c.CertDirectory, "cert-dir", c.CertDirectory, "The directory where the TLS certs are located. "+
-		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
 
 	fs.StringVar(&c.RootDirectory, "root-dir", c.RootDirectory, "Directory path for managing kubelet files (volume mounts,etc).")
```

Comment:
There is no compatibility guarantee, as it's an alpha object (and generally there are no compatibility guarantees if you run with any alpha feature gate enabled).
Yeah, it should be in flags too. I thought it already was but I'm probably just remembering moving it in #44252. I'll move it.

Thanks, looks like I documented the resolution behavior in a comment but not in the help text.